src/main/java/frc/utils/auto/PathPlannerUtils.java

```diff
@@ -0,0 +1,94 @@
+package frc.utils.auto;
+
+import com.pathplanner.lib.auto.AutoBuilder;
+import com.pathplanner.lib.auto.NamedCommands;
+import com.pathplanner.lib.commands.PathfindingCommand;
+import com.pathplanner.lib.controllers.PPHolonomicDriveController;
+import com.pathplanner.lib.path.GoalEndState;
+import com.pathplanner.lib.path.PathConstraints;
+import com.pathplanner.lib.path.PathPlannerPath;
+import com.pathplanner.lib.pathfinding.Pathfinding;
+import com.pathplanner.lib.util.HolonomicPathFollowerConfig;
+import edu.wpi.first.math.Pair;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Subsystem;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.BooleanSupplier;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+public class PathPlannerUtils {
+
+    public static void startPathPlanner() {
+        makePathPlannerCompatibleWithAdvantageKit();
+        scheduleWarmup();
+    }
+
+    private static void makePathPlannerCompatibleWithAdvantageKit() {
+        Pathfinding.setPathfinder(new LocalADStarAK());
+    }
+
+    private static void scheduleWarmup() {
+        PathfindingCommand.warmupCommand().schedule();
+    }
+
+    public static void configurePathPlanner(
+            Supplier<Pose2d> poseSupplier,
+            Consumer<Pose2d> resetPose,
+            Supplier<ChassisSpeeds> robotRelativeSpeedsSupplier,
+            Consumer<ChassisSpeeds> robotRelativeOutput,
+            HolonomicPathFollowerConfig config,
+            BooleanSupplier shouldFlipPath,
+            Subsystem driveSubsystem
+    ) {
+        AutoBuilder.configureHolonomic(
+                poseSupplier,
+                resetPose,//todo - maybe cancel, if vision very accurate solving wrong robot placement
```

Comment:
ask amir

Reply:
delete the comment, if the people don't want to reset they will give supplier that do nothing